### PR TITLE
should unroll the outer dim for 2D pointwiser scheduler 

### PR DESCRIPTION
**Pull request by liqiangxl**
**Issue:** The 2D pointwise scheduler is used to save loaded data size when there are outer broadcast TensorViews. It should unroll along the outer dimension to reuse data across different rows. The previous implementation mistakenly unrolled the inner dimension.

**Fix:**
(1) Fix unroll dim
(2) Test number of blocks is reduced by unroll factor

### CODE REVIEW

File: csrc/scheduler/pointwise.cpp [hunk 1]
```diff
@@ -704,16 +704,19 @@ void schedulePointwise(Fusion* fusion, const PointwiseParams* pparams) {
       if (pparams->vectorization_factor > 1) {
         reference_tv->split(1, pparams->vectorization_factor);
       }
-      // [outer | inner/vect, vect]
+      // [outer | i-remainder, Vect]
       reference_tv->split(1, NamedScalar::getParallelDim(ParallelType::TIDx));
-      // [outer | inner/vect/TIDx, TIDx, vect]
-      reference_tv->split(1, pparams->unroll_factor);
```
Comment (liqiangxl) [inline csrc/scheduler/pointwise.cpp:710]: Here, shoud unroll outer dim.

**Review by rdspring1 — APPROVED** [review]
LGTM.

Should we add a comment for the `unroll_factor` argument?
https://github.com/NVIDIA/Fuser/blob/main/csrc/scheduler/pointwise_heuristic.h#L44-L45

**Comment by liqiangxl** [comment]
> LGTM.
> 
> Should we add a comment for the `unroll_factor` argument? https://github.com/NVIDIA/Fuser/blob/main/csrc/scheduler/pointwise_heuristic.h#L44-L45

added
```
  // Unroll on top of vectorization
  // In the 2D scheduler, unroll the outer dimension to reuse loaded data across
  // rows, reducing loaded bytes by the unroll factor.
```

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
> **Issue:** The 2D pointwise scheduler is used to save loaded data size when there are outer broadcast TensorViews. It should unroll along the outer dimension to reuse data across different rows. The previous implementation mistakenly unrolled the inner dimension.
> 
> **Fix:** (1) Fix unroll dim (2) Test number of blocks is reduced by unroll factor

Actually, the original implementation should be used when we have inner dim broadcast. The best position to unroll depends on the actual problem and should be set by the heuristics.

**Pull request merged**